tests/recipes/recipe_lib_test.py

```diff
@@ -0,0 +1,159 @@
+    """
+
+    def __new__(cls, *args):
+        obj = super().__new__(cls)
+        if obj.recipe_name is not None:
+            print(f"We are testing recipe: {obj.recipe_name}")
+            obj.recipes.append(obj.recipe_name)
+            obj.recipe_build_order.insert(1, obj.recipe_name)
+            return obj
+
+    @mock.patch("pythonforandroid.recipe.Recipe.check_recipe_choices")
+    @mock.patch("pythonforandroid.build.ensure_dir")
+    @mock.patch("pythonforandroid.archs.glob")
+    @mock.patch("pythonforandroid.archs.find_executable")
+    def test_get_recipe_env(
+        self,
+        mock_find_executable,
+        mock_glob,
+        mock_ensure_dir,
+        mock_check_recipe_choices,
+    ):
+        """
+        Test that get_recipe_env contains some expected arch flags and that
+        some internal methods has been called.
+        """
+        mock_find_executable.return_value = self.expected_compiler.format(
+                android_ndk=self.ctx._ndk_dir
+        )
+        mock_glob.return_value = ["llvm"]
+        mock_check_recipe_choices.return_value = sorted(
+            self.ctx.recipe_build_order
+        )
+
+        # make sure the arch flags are in env
+        env = self.recipe.get_recipe_env(self.arch)
+        for flag in self.arch.arch_cflags:
+            self.assertIn(flag, env["CFLAGS"])
+        self.assertIn(
+            f"-target {self.arch.target}",
+            env["CFLAGS"],
+        )
+
+        for flag, value in self.extra_env_flags.items():
+            self.assertIn(value, env[flag])
+
+        # make sure that the mocked methods are actually called
+        mock_glob.assert_called()
+        mock_ensure_dir.assert_called()
+        mock_find_executable.assert_called()
+        mock_check_recipe_choices.assert_called()
+
+    @mock.patch("pythonforandroid.util.chdir")
+    @mock.patch("pythonforandroid.build.ensure_dir")
+    @mock.patch("pythonforandroid.archs.glob")
+    @mock.patch("pythonforandroid.archs.find_executable")
+    def test_build_arch(
+        self,
+        mock_find_executable,
+        mock_glob,
+        mock_ensure_dir,
+        mock_current_directory,
+    ):
+        mock_find_executable.return_value = self.expected_compiler.format(
+                android_ndk=self.ctx._ndk_dir
+        )
+        mock_glob.return_value = ["llvm"]
+
+        # Since the following mocks are dynamic,
+        # we mock/test it inside a Context Manager
+        with mock.patch(
+            f"pythonforandroid.recipes.{self.recipe_name}.sh.Command"
+        ) as mock_sh_command, mock.patch(
+            f"pythonforandroid.recipes.{self.recipe_name}.sh.make"
+        ) as mock_make:
+            self.recipe.build_arch(self.arch)
+
+            for command in self.sh_command_calls:
+                self.assertIn(
+                    mock.call(command),
+                    mock_sh_command.mock_calls,
+                )
+            mock_make.assert_called()
```

Comment:
FYI actually this check can also be outside the context manager. The only thing that needs to be in the context manager is the call that you want to be mocked, hence `self.recipe.build_arch(self.arch)`